### PR TITLE
docs: updated content via Joe's blog post on "Common Good Parachains"

### LINKED ISSUE #1827: Add more info about common good parachains
Based on Joe's article - https://polkadot.network/common-good-parachains-an-introduction-to-governance-allocated-parachain-slots/

### PR DESCRIPTION
**Pull request by jakehemmerle**
Paraphrased and copied  @joepetrowski 's awesome common good parachain article into the wiki 

Should close #1827

### CODE REVIEW

File: docs/learn-parachains.md [hunk 1]
```diff
@@ -79,6 +80,71 @@ Reserved balances are non-transferrable and cannot be used for staking. At the e
...
+DOT-denominated smart contract platforms. All of these could operate without a new token:
+
+A bridge could add its own native token to charge as a toll, but in many cases that would be
+arbitrary value capture, when it could just as well use DOT and/or the bridged chain’s assets in its
+fee mechanism. A DOT-denominated smart contract layer-one blockchain would allow Wasm smart contract
+execution using DOT as the native asset with which to pay gas. A generic assets chain would allow
+anyone to place a deposit in DOT to deploy their asset on-chain. Assets on this chain could be
+backed by physical goods like artwork, real estate, or gold; or by paper goods like shares in a
+company or fiat currency held by a trusted party, providing a stable, permanent launchpad for
+stablecoins and Central Bank Digital Currencies. Public utility parachains would typically grant
```
Comment (joepetrowski) [inline docs/learn-parachains.md]:
```suggestion
stablecoins and Central Bank Digital Currencies.

Public utility parachains would typically grant
```

File: docs/learn-parachains.md [hunk 2]
```diff
@@ -79,6 +80,71 @@ Reserved balances are non-transferrable and cannot be used for staking. At the e
...
+Some examples of potential public utility chains are bridges, generic asset chains, and
+DOT-denominated smart contract platforms. All of these could operate without a new token:
+
+A bridge could add its own native token to charge as a toll, but in many cases that would be
+arbitrary value capture, when it could just as well use DOT and/or the bridged chain’s assets in its
+fee mechanism. A DOT-denominated smart contract layer-one blockchain would allow Wasm smart contract
+execution using DOT as the native asset with which to pay gas. A generic assets chain would allow
+anyone to place a deposit in DOT to deploy their asset on-chain. Assets on this chain could be
+backed by physical goods like artwork, real estate, or gold; or by paper goods like shares in a
+company or fiat currency held by a trusted party, providing a stable, permanent launchpad for
```
Comment (joepetrowski) [inline docs/learn-parachains.md]: Maybe make these bullet points?

File: docs/learn-parachains.md [hunk 3]
```diff
@@ -79,6 +80,71 @@ Reserved balances are non-transferrable and cannot be used for staking. At the e
 DOT is unreserved. Parachains that have not secured a new lease to extend their slot will
 automatically become parathreads.
 
+## Common Good Parachains
+
+"Common Good" parachains are parachain slots reserved for functionality that benefits the the Web3
```
Comment (joepetrowski) [inline docs/learn-parachains.md]: Web3 is a bit broad here. The main point is that they help other layer 1 blockchains operate more effectively by reducing repeated logic that everyone would want to use. Having more efficient parachains benefits Web3, but that's kind of one step removed here.
```suggestion
"Common Good" parachains are parachain slots reserved for functionality that benefits the the Polkadot
```

File: docs/learn-parachains.md [hunk 4]
```diff
@@ -79,6 +80,71 @@ Reserved balances are non-transferrable and cannot be used for staking. At the e
...
+execution using DOT as the native asset with which to pay gas. A generic assets chain would allow
+anyone to place a deposit in DOT to deploy their asset on-chain. Assets on this chain could be
+backed by physical goods like artwork, real estate, or gold; or by paper goods like shares in a
+company or fiat currency held by a trusted party, providing a stable, permanent launchpad for
+stablecoins and Central Bank Digital Currencies. Public utility parachains would typically grant
+privileged business logic to Polkadot’s governance. Just as the Polkadot Relay Chain has several
+privileged functions like setting the validator count or allocating DOT from the Treasury, these
+parachains can have privileged functions like changing system parameters or registering an asset.
+
+Because public utility chains add functionality beyond the scope of the Relay Chain, we expect the
```
Comment (joepetrowski) [inline docs/learn-parachains.md]: We don't normally use first person in the Wiki, and here unfortunately it leads us into passive voice, like: "... public utility chains will likely be approved only in rare scenarios."

**Review by joepetrowski — COMMENTED** [review]
Looks good, just a few tweaks.

**Review by Swader — APPROVED** [review]

File: docs/learn-parachains.md (hunk 4 above)
Comment (jakehemmerle) [reply docs/learn-parachains.md]: I forgot 'we' is still first person!

**Pull request merged**